### ISSUE TITLE
Pact-ffi response is out of order

### ISSUE THREAD

**Issue by CraigRichards** [pact-foundation/pact-reference#168]
I've recently gone through this example and noticed the output is not ordered correctly. All the Givens are listed first rather than being associated with their scenario. Here is an example output.

#9 121.0 Verifying a pact between ApiClient and ProductService
#9 121.0   Given no products exist
#9 121.1   Given product with ID 10 exists
#9 121.2   Given product with ID 11 does not exist
#9 121.2   Given product with ID 9 exists
#9 121.2   Given products exist
#9 121.2   A valid request for a product and none exist
#9 121.2     returns a response which
#9 121.2       has status code 200 (OK)
#9 121.2       includes headers
#9 121.2         "Content-Type" with value "application/json; charset=utf-8" (OK)
#9 121.2       has a matching body (OK)
#9 121.2   A valid request for product 10
#9 121.2     returns a response which
#9 121.2       has status code 200 (OK)
#9 121.2       includes headers
#9 121.2         "Content-Type" with value "application/json; charset=utf-8" (OK)
#9 121.2       has a matching body (OK)
#9 121.2   A valid request for a product 11 that does not exist
#9 121.2     returns a response which
#9 121.2       has status code 404 (OK)
#9 121.2       has a matching body (OK)
#9 121.2   A valid request for product 9
#9 121.2     returns a response which
#9 121.2       has status code 200 (OK)
#9 121.2       includes headers
#9 121.2         "Content-Type" with value "application/json; charset=utf-8" (OK)
#9 121.2       has a matching body (OK)
#9 121.2   A valid request for all products
#9 121.2     returns a response which
#9 121.2       has status code 200 (OK)
#9 121.2       includes headers
#9 121.2         "Content-Type" with value "application/json; charset=utf-8" (OK)
#9 121.2       has a matching body (OK)

**Comment by mefellows** [comment]
Thanks @CraigRichards. I've tested this with the latest verifier CLI and the issue is definitely in the core, and not in the language bindings.

e.g. with an example pact I have in the pact js repo:

```
pact_verifier_cli-osx-x86_64 --loglevel none --state-change-url http://localhost:57400/_pactSetup --provider-name vehicle-mission-sync --port 57400 --hostname localhost --user undefined --password undefined --provider-version 1 --file ./pacts/TransactionService-AccountService.json

Verifying a pact between TransactionService and AccountService
  Given Account Test001 exists
  Given Account Test001 exists
  Given set id

  a request to get the account details
     ...

  a request to get the account details via POST
      ...

  a request to get the plain data
      ...
```

**Comment by rholshausen** [comment]
Fixed with commit 6cae9b0932
